Java 7 and Jenkins 1.625.1 are now required

aheritier opened the pull request:
Maven 3.3.x requires also Java 7
Jenkins 1.625.1 is the first LTS which requires Java 7 (see also https://jenkins.io/blog/2015/04/06/good-bye-java6/ )
Java 6 support for Maven jobs is already impossible for Jenkins >= 1.612

aheritier commented:
please @reviewbybees @jenkinsci/code-reviewers @olamy

KostyaSha commented:
👍

reviewbybees commented:
This pull request originates from a [CloudBees](https://www.cloudbees.com/) employee. At CloudBees, we require that all pull requests be reviewed by other CloudBees employees before we seek to have the change accepted. If you want to learn more about our process please see [this explanation](https://github.com/reviewbybees/about#about-reviewbybees).

aheritier commented:
Do you think that we should bump the version to the next major ? 3.x ? cc @KostyaSha

KostyaSha commented:
Do you mean 1.xxx.3? Or maven 3.x?

aheritier commented:
The plugin version @KostyaSha 
For now we will release it (one day) as 2.14
Maybe we should go to 3.0 to show that there is the Java compatibility change (+ it will fix the support for Maven 3.3.x)

KostyaSha commented:
No objections.

jglick commented:
```
java.lang.AssertionError: expected:<2> but was:<null>
	at hudson.maven.MavenMultiModuleTest.symlinksUpdated(MavenMultiModuleTest.java:73)
```

jglick commented:
:bee: assuming the test failure is ironed out.

aheritier commented:
@jglick crap I didn't have this test failure locally. I will verify if I missed something.

oleg-nenashev commented:
🐝

aheritier commented:
Hi @jglick (cc @olamy @oleg-nenashev @KostyaSha @batmat ) I fixed the failing tests. In fact I removed what I consider to be 2 wrong assertions ( #23a2ae2 ) but I'm not sure why they were ok previously (perhaps because of https://github.com/jenkinsci/jenkins/commit/99c43c8ccd7b748941c9516aabf1c78c0e2b6491 ) 
Let me know if you have any feedback. Thanks

aheritier commented:
@reviewbybees done
